snmp/assets/configuration/spec.yaml

```diff
@@ -552,3 +552,9 @@ files:
       - template: instances/default
```

Comment:
A tiny bit unrelated to this PR, but noticing as I'm reviewing it

I'm not sure when `metric_patterns` was added on that config file this but I don't think we support it especially as snmp is a go check.
I think we should be able to remove it by having
```
overrides:
  metric_patterns.hidden: true
```

Reply:
https://github.com/DataDog/integrations-core/pull/17407/commits/73c0f9cb7fa74cd2ed8c3780a98a17e27e0d92cf is that ok for you ?